src/libOpenImageIO/imagebufalgo_opencv.cpp

```diff
@@ -136,7 +137,182 @@ IplImage *
+
+            if (!tmp.get_pixels(ROI::All(), TypeDesc::UINT16, &data[0])) {
+                DASSERT (0 && "Could not get pixels from ImageBuf.");
+                return NULL;
+            }
+
+            cvSetData(dst, &data[0], dst->widthStep);
+
+            break;
+        }
+        case TypeDesc::INT16: {
+            dst = cvCreateImageHeader(size, IPL_DEPTH_16S, spec.nchannels);
+
+            if (!dst) {
+                DASSERT (0 && "Could not create dst IplImage.");
+                return NULL;
+            }
+
+            std::vector<signed int> data;
+            data.resize(spec.width * spec.height * spec.depth * spec.nchannels);
+
+            if (!tmp.get_pixels(ROI::All(), TypeDesc::INT16, &data[0])) {
+                DASSERT (0 && "Could not get pixels from ImageBuf.");
+                return NULL;
+            }
+
+            cvSetData(dst, &data[0], dst->widthStep);
+
+            break;
+        }
+        // OpenCV doesn't have a half type, so promote to float.
+        case TypeDesc::HALF: {
+            dst = cvCreateImageHeader(size, IPL_DEPTH_32F, spec.nchannels);
+
+            if (!dst) {
+                DASSERT (0 && "Could not create dst IplImage.");
+                return NULL;
+            }
+
+            std::vector<float> data;
+            data.resize(spec.width * spec.height * spec.depth * spec.nchannels);
+
+            if (!tmp.get_pixels(ROI::All(), TypeDesc::FLOAT, &data[0])) {
+                DASSERT (0 && "Could not get pixels from ImageBuf.");
+                return NULL;
+            }
+
+            cvSetData(dst, &data[0], dst->widthStep);
+
+            break;
+        }
+        case TypeDesc::FLOAT: {
+            dst = cvCreateImageHeader(size, IPL_DEPTH_32F, spec.nchannels);
+
+            if (!dst) {
+                DASSERT (0 && "Could not create dst IplImage.");
+                return NULL;
+            }
+
+            std::vector<float> data;
+            data.resize(spec.width * spec.height * spec.depth * spec.nchannels);
+
+            if (!tmp.get_pixels(ROI::All(), TypeDesc::FLOAT, &data[0])) {
+                DASSERT (0 && "Could not get pixels from ImageBuf.");
+                return NULL;
+            }
+
+            cvSetData(dst, &data[0], dst->widthStep);
+
+            break;
+        }
+        case TypeDesc::DOUBLE: {
+            dst = cvCreateImageHeader(size, IPL_DEPTH_64F, spec.nchannels);
+
+            if (!dst) {
+                DASSERT (0 && "Could not create dst IplImage.");
+                return NULL;
+            }
+
+            std::vector<double> data;
+            data.resize(spec.width * spec.height * spec.depth * spec.nchannels);
+
+            if (!tmp.get_pixels(ROI::All(), TypeDesc::DOUBLE, &data[0])) {
+                DASSERT (0 && "Could not get pixels from ImageBuf.");
+                return NULL;
+            }
+
+            cvSetData(dst, &data[0], dst->widthStep);
+
+            break;
+        }
+        default:
```

Comment:
Maybe for the `default` case, rather than making an error, you should just have `default:` also direct to the UINT8 case (or the FLOAT case?), meaning that if it's not a supported type, just convert to float and keep chugging?